sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -28,6 +28,22 @@ import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
 
 object IntervalUtils {
 
+  object IntervalUnit extends Enumeration {
```

Comment:
This may be an old idea, but I thought the java Enum class was the preferred way to define enums in scala, even?

Reply:
I haven't found this requirement in the Scala coding guide. Where does it come from (do you have some links for related discussions)?

Reply:
At least, there are 46 places where Scala's `Numeration` is used:
```shell
$ find . -name "*.scala" -print0|xargs -0 grep -n 'extends Enumeration'|wc -l
      46
```

Reply:
It's stuff like: https://groups.google.com/forum/#!topic/scala-internals/8RWkccSRBxQ%5B1-25%5D

But indeed we use scala.Enumeration widely, and I think the issues crop up when interoperating with Java, or in more obscure cases. I think it's fine here.